src/profile-logic/marker-schema.js

```diff
@@ -0,0 +1,217 @@
+      { key: 'functionLine', label: 'Function Line', format: 'integer' },
+      { key: 'bailoutLine', label: 'Bailout Line', format: 'integer' },
+    ],
+  },
+  {
+    name: 'GCMajor',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [
+      // Use custom handling
+    ],
+  },
+  {
+    name: 'GCMinor',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [
+      // Use custom handling
+    ],
+  },
+  {
+    name: 'GCSlice',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [
+      // Use custom handling
+    ],
+  },
+  {
+    name: 'CC',
+    label: 'Cycle Collect',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [],
+  },
+  {
+    name: 'FileIO',
+    display: [],
+    data: [
+      {
+        key: 'operation',
+        label: 'Operation',
+        format: 'string',
+        searchable: true,
+      },
+      {
+        key: 'source',
+        label: 'Source',
+        format: 'string',
+        searchable: true,
+      },
+      {
+        key: 'filename',
+        label: 'Filename',
+        format: 'file-path',
+        searchable: true,
+      },
+    ],
+  },
+  {
+    name: 'MediaSample',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      {
+        key: 'sampleStartTimeUs',
+        label: 'Sample start time',
+        format: 'microseconds',
+      },
+      {
+        key: 'sampleEndTimeUs',
+        label: 'Sample end time',
+        format: 'microseconds',
+      },
+    ],
+  },
+  {
+    name: 'Styles',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      {
+        key: 'elementsTraversed',
+        label: 'Elements traversed',
+        format: 'integer',
+      },
+      { key: 'elementsStyled', label: 'Elements styled', format: 'integer' },
+      { key: 'elementsMatched', label: 'Elements matched', format: 'integer' },
+      { key: 'stylesShared', label: 'Styles shared', format: 'integer' },
+      { key: 'stylesReused', label: 'Styles reused', format: 'integer' },
+    ],
+  },
+  {
+    name: 'PreferenceRead',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'prefName', label: 'Name', format: 'string' },
+      { key: 'prefKind', label: 'Kind', format: 'string' },
+      { key: 'prefType', label: 'Type', format: 'string' },
+      { key: 'prefValue', label: 'Value', format: 'string' },
+    ],
+  },
+  {
+    name: 'Invalidation',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'url', label: 'URL', format: 'url' },
+      { key: 'line', label: 'Line', format: 'integer' },
+    ],
+  },
+  {
+    name: 'UserTiming',
+    display: ['marker-chart', 'marker-table'],
+    data: [{ key: 'name', label: 'Name', format: 'string' }],
+  },
+  {
+    name: 'Text',
+    display: ['marker-chart', 'marker-table'],
+    data: [{ key: 'name', label: 'Name', format: 'string' }],
```

Comment:
Not for this PR, but I really don't like `Name` for  this field. This is not a `name` technically, but rather either `data` or directly `text`. I hope we can change this. (ofc end goal is to stop using this 🙂)

Reply:
I think we can change it here. How about "Details" instead. It's nice and generic. This marker is used in a really surprising ways.

Reply:
Looks good to me!